WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostListAdapter.kt

```diff
@@ -101,7 +103,17 @@ class PostListAdapter(
         notifyDataSetChanged()
     }
 
-    private class LoadingViewHolder(view: View) : ViewHolder(view)
+    private class LoadingViewHolder(view: View, postListType: PostListType) : ViewHolder(view) {
+        init {
```

Comment:
I think we might want to consider moving this business logic out of the view layer. I realize this might sound like a nitpick and it might look like it's not worth the effort. However, the code isn't testable at the moment and imho if we start making exception it'll be difficult to find the line what is ok to be an exception and what is not.

I'd consider one of the following options
1. Change the name of the actions on the loading items to something generic so they can be used across all the post list types.

2. Create a new state for LoadingTrashedItem [here](https://github.com/wordpress-mobile/WordPress-Android/blob/006ece361213509b5b529367508dfd37b551969e/WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListItemType.kt#L18) and call something like "createLoadingState(..)" lambda [here](https://github.com/wordpress-mobile/WordPress-Android/blob/f13928adaf0928f09be0974ffc873aa3887be6af/WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListItemDataSource.kt#L97). 

3. Create a LoadingUIState with "showEditButton, showXyzButton, ..." fields and call something like "createLoadingState(..)" lambda [here](https://github.com/wordpress-mobile/WordPress-Android/blob/f13928adaf0928f09be0974ffc873aa3887be6af/WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListItemDataSource.kt#L97).

4. Combine option 2 and 3 -> introduce object `LoadingTrashedItem` with constant fields "showEditButton=false,..., showMoveToDraftButton = true, ..." and add the same fields with opposite values to `LoadingItem` so we can reuse the same layout file for both loading states.

If you think it's not worth the effort, I'd at least suggest adding a comment that this is an exception and more business logic shouldn't be added to the view layer.

Reply:
I moved the logic out the view holder in `b78a51d`, by introducing a `LoadingItemOptions` class that controls what we display in the `LoadingViewHolder`

Reply:
I moved the logic out the view holder in b78a51d, by introducing a `LoadingItemOptions` class that controls what we display in the `LoadingViewHolder`